firmware/controllers/math/speed_density.cpp

```diff
@@ -29,42 +29,48 @@ baroCorr_Map3D_t baroCorrMap("baro");
 #define tpMin 0
 #define tpMax 100
 //  http://rusefi.com/math/t_charge.html
+/***panel:Charge Temperature*/
 float getTCharge(int rpm, float tps, float coolantTemp, float airTemp DECLARE_ENGINE_PARAMETER_SUFFIX) {
 	if (cisnan(coolantTemp) || cisnan(airTemp)) {
 		warning(CUSTOM_ERR_NAN_TCHARGE, "t-getTCharge NaN");
 		return coolantTemp;
 	}
 
-	float Tcharge_coff;
 
-	if (CONFIG(tChargeMode) == TCHARGE_MODE_AIR_INTERP) {
+	if ((engine->engineState./*DISPLAY_IF*/isTChargeAirModel = (CONFIG(tChargeMode) == TCHARGE_MODE_AIR_INTERP)) == true) {
```

Comment:
`== true` should never be typed - just omit it and you'll get the same behavior

Reply:
You are right, this was part of a wrong "error: suggest parentheses around assignment used as truth value -Werror=parentheses" fix. Will change now.

Reply:
oh - let's also avoid assignment in an if statement then :smile:

Reply:
need to save condition check result for live docs. Did not want to move condition logic from if statement, it would get ugly if we have chained if/else if